onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -0,0 +1,571 @@
+        }
+      }
+
+      auto iter = node->OutputNodesBegin();
+
+      if (iter == node->OutputNodesEnd()) {
+        return false;
+      }
+
+      for (auto it = node->OutputNodesBegin(); it != node->OutputNodesEnd(); ++it) {
+        auto out_node = graph_viewer.GetNode((*it).Index());
+
+        if (out_node->OpType() != "Add") {
+          return false;
+        }
+      }
+
+      bool is_graph_input = false;
+
+      auto graph_inputs = graph_viewer.GetInputs();
+      for (const auto& input : node->InputDefs()) {
+        auto it = find(graph_inputs.begin(), graph_inputs.end(), input);
+        if (it != graph_inputs.end()) {
+          is_graph_input = true;
+        }
+      }
+      if (is_graph_input) {
+        size_t input_dims_size = input_arrays[0].size();
+
+        //Only support 2D Matmul operations
+        for (int i = 0; i < num_inputs; i++) {
+          if (input_arrays[i].size() > 2)
+            return false;
+
+          if (input_dims_size != input_arrays[i].size())
+            return false;
+        }
+      }
+
+      //Disable MNIST for MYRIAD and HDDL
+      if (dev_id == "MYRIAD" || dev_id == "HDDL")
+        return false;
+    }
+    //Dropout or Identity can't have graph inputs
+    if (node->OpType() == "Dropout" || node->OpType() == "Identity" || node->OpType() == "Concat") {
+      auto graph_inputs = graph_viewer.GetInputs();
+      for (auto input : node->InputDefs()) {
+        auto it = find(graph_inputs.begin(), graph_inputs.end(), input);
+        if (it != graph_inputs.end()) {
+          return false;
+        }
+      }
+    }
+
+    //Attribute auto pad for MaxPool and AveragePool must not be empty or SAME_LOWER
+    if (node->OpType() == "MaxPool" || node->OpType() == "AveragePool") {
+      auto attributes = node->GetAttributes();
+      auto auto_pad = attributes["auto_pad"].s();
+      if (auto_pad == "" || auto_pad == "SAME_LOWER"){
+        return false;
+      }
+
+      auto strides_ints = attributes["strides"].ints();
+      if(auto_pad == "SAME_UPPER" && strides_ints.size() == 0){
+          return false;
+      }
+
+      //Dilations have to be 1
+      auto dilations_ints = attributes["dilations"].ints();
+      if (dilations_ints.size() != 0) {
+        if (dilations_ints[0] > 1)
+          return false;
+      }
+
+      //Don't support ceil_mode = 1
+      auto ceil_mode = attributes["ceil_mode"].i();
+      if (ceil_mode != 0)
+        return false;
+
+      //Don't support multiple outputs for Pooling
+      if (node->OutputDefs().size() > 1) {
+        return false;
+      }
+    }
+
+    //Only support 4D and 5D blobs for CPU,GPU
+    //Only support 3D and 4D blobs for MYRIAD and HDDL
+    if (node->OpType() == "GlobalMaxPool" || node->OpType() == "MaxPool" || node->OpType() == "AveragePool" || node->OpType() == "GlobalAveragePool") {
+      auto graph_inputs = graph_viewer.GetInputs();
+      auto it = find(graph_inputs.begin(), graph_inputs.end(), node->InputDefs()[0]);
```

Comment:
not sure i'm following this logic.
this check only happens if the this node input is the same as graph input? 
and what if it's not?

Reply:
did you mean to do something like this instead: 
https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.h#L110

Reply:
Hey George,

Yes i was trying to figure out a way to get the input dimensions of an internal node but was running into a seg fault. I thought that at GetCapability the shapes are not determined and will be done at compile using shape inference.

Reply:
I will use this logic